Source/WebKit/WebProcess/GPU/media/AudioVideoRendererRemote.cpp

```diff
@@ -0,0 +1,839 @@
+    if (!isGPURunning() || !gpuProcessConnection)
+        return;
+
+    gpuProcessConnection->connection().send(Messages::RemoteAudioVideoRendererProxyManager::SetShouldDisableHDR(m_identifier, disable), 0);
+}
+
+void AudioVideoRendererRemote::setPlatformDynamicRangeLimit(const PlatformDynamicRangeLimit& limit)
+{
+    RefPtr gpuProcessConnection = m_gpuProcessConnection.get();
+    if (!isGPURunning() || !gpuProcessConnection)
+        return;
+
+    gpuProcessConnection->connection().send(Messages::RemoteAudioVideoRendererProxyManager::SetPlatformDynamicRangeLimit(m_identifier, limit), 0);
+}
+
+void AudioVideoRendererRemote::setResourceOwner(const ProcessIdentity& processIdentity)
+{
+    RefPtr gpuProcessConnection = m_gpuProcessConnection.get();
+    if (!isGPURunning() || !gpuProcessConnection)
+        return;
+
+
+    gpuProcessConnection->connection().send(Messages::RemoteAudioVideoRendererProxyManager::SetResourceOwner(m_identifier, ProcessIdentity { processIdentity }), 0);
+}
+
+void AudioVideoRendererRemote::flushAndRemoveImage()
+{
+    RefPtr gpuProcessConnection = m_gpuProcessConnection.get();
+    if (!isGPURunning() || !gpuProcessConnection)
+        return;
+
+    gpuProcessConnection->connection().send(Messages::RemoteAudioVideoRendererProxyManager::FlushAndRemoveImage(m_identifier), 0);
+}
+
+RefPtr<VideoFrame> AudioVideoRendererRemote::currentVideoFrame() const
+{
+    RefPtr gpuProcessConnection = m_gpuProcessConnection.get();
+    if (!isGPURunning() || !gpuProcessConnection)
+        return nullptr;
+
+    auto sendResult = gpuProcessConnection->connection().sendSync(Messages::RemoteAudioVideoRendererProxyManager::CurrentVideoFrame(m_identifier), 0);
+    if (!sendResult.succeeded())
+        return nullptr;
+
+    auto [result] = sendResult.takeReply();
+    if (result)
+        return RemoteVideoFrameProxy::create(gpuProcessConnection->connection(), gpuProcessConnection->protectedVideoFrameObjectHeapProxy(), WTFMove(*result));
+    return nullptr;
+}
+
+std::optional<VideoPlaybackQualityMetrics> AudioVideoRendererRemote::videoPlaybackQualityMetrics()
+{
+    return m_state.videoPlaybackQualityMetrics;
+}
+
+PlatformLayer* AudioVideoRendererRemote::platformVideoLayer() const
+{
+#if PLATFORM(COCOA)
+    if (!m_videoLayer && m_layerHostingContext.contextID) {
+        auto expandedVideoLayerSize = expandedIntSize(videoLayerSize());
+        m_videoLayer = createVideoLayerRemote(const_cast<AudioVideoRendererRemote&>(*this), m_layerHostingContext.contextID, WebCore::MediaPlayer::VideoGravity::ResizeAspect, expandedVideoLayerSize);
+        m_videoLayerManager->setVideoLayer(m_videoLayer.get(), expandedVideoLayerSize);
+    }
+    return m_videoLayerManager->videoInlineLayer();
+#else
+    return nullptr;
+#endif
+}
+
+#if ENABLE(VIDEO_PRESENTATION_MODE)
+void AudioVideoRendererRemote::setVideoFullscreenLayer(PlatformLayer* videoFullscreenLayer, WTF::Function<void()>&& completionHandler)
```

Comment:
Could completionHandler be a CompletionHandler?

Reply:
that callback comes from the HTMLMediaElement `WEBCORE_EXPORT void setVideoFullscreenLayer(PlatformLayer*, Function<void()>&& completionHandler = [] { });` it probably should be